AnnService/inc/Core/Common/WorkSpace.h

```diff
@@ -93,17 +96,32 @@ namespace SPTAG
                 }
             }
 
+            inline int HashTableExponent() const { return m_exp; }
 
             inline bool CheckAndSet(SizeType idx)
             {
                 // Inner Index is begin from 1
-                return _CheckAndSet(m_hashTable.get(), idx + 1) == 0;
+                return _CheckAndSet(m_hashTable.get(), m_poolSize, true, idx + 1) == 0;
             }
 
+            inline void DoubleSize()
+            {
+                int new_poolSize = ((m_poolSize + 1) << 1) - 1; 
+                SizeType* new_hashTable = new SizeType[(new_poolSize + 1) * 2];
+                memset(new_hashTable, 0, sizeof(SizeType) * (new_poolSize + 1) * 2);
+
+                m_secondHash = false;
+                for (int i = 0; i <= new_poolSize; i++)
+                    if (m_hashTable[i]) _CheckAndSet(new_hashTable, new_poolSize, true, m_hashTable[i]);
+
+                m_exp++;
+                m_poolSize = new_poolSize;
+                m_hashTable.reset(new_hashTable);
```

Comment:
just confirm the original pool size be released correctly to avoid memory leak

Reply:
It will be released correctly.